components/style/properties/properties.mako.rs

```diff
@@ -687,18 +687,42 @@ impl LonghandId {
     fn is_early_property(&self) -> bool {
         matches!(*self,
             % if product == 'gecko':
-            LonghandId::TextOrientation |
+            // We need to know the number of animations / transition-properties
+            // before setting the rest of the related longhands, see #15923.
+            //
+            // FIXME(emilio): Looks to me that we could just do this in Gecko
+            // instead of making them early properties. Indeed, the spec
+            // mentions _used_ values, not computed values, so this looks wrong.
             LonghandId::AnimationName |
             LonghandId::TransitionProperty |
-            LonghandId::XLang |
+
+            // FIXME(emilio): why?
+            LonghandId::TextOrientation |
+
+            // Needed to properly compute the zoomed font-size.
+            //
+            // FIXME(emilio): This could probably just be a cascade flag like
+            // IN_SVG_SUBTREE or such, and we could nuke this property.
             LonghandId::XTextZoom |
+
+            // Needed to do font-size computation in a language-dependent way.
+            //
+            // FIXME(emilio): The two MathML props add a lot of complexity, it'd
+            // be nice to nuke them.
+            LonghandId::XLang |
             LonghandId::MozScriptLevel |
             LonghandId::MozMinFontSizeRatio |
```

Comment:
This is not really for MathML prop, IIRC. It was for ruby annotation to have a different min font size.

Also we are probably adding `font-{min,max}-size` or something similar to the spec, so we may end up extending them here (rather than nuke it).